node/src/rpc.rs

```diff
@@ -44,26 +38,12 @@ pub struct FullDeps<C, P> {
 }
 
 /// Instantiate all RPC extensions.
-pub fn create_full_pendulum<C, P>(
-	deps: FullDeps<C, P>,
+pub fn create_full_pendulum<P>(
```

Comment:
All the functions in this file were unnecessarily generic. This is not required – they are used for exactly one of the four runtimes. Thus, also the complex type guards could be removed.